metis-debias/metis-debias-detect-service/src/main/java/eu/europeana/metis/debias/detect/client/DeBiasClient.java

```diff
@@ -0,0 +1,121 @@
+    clientHttpRequestFactory.setConnectionRequestTimeout(this.requestTimeout);
+    final RestClient restClient = RestClient.builder()
+                                            .messageConverters(httpMessageConverters -> {
+                                              httpMessageConverters.add(new StringHttpMessageConverter());
+                                              httpMessageConverters.add(new MappingJackson2HttpMessageConverter());
+                                            })
+                                            .build();
+
+    final ResponseEntity<DeBiasResult> response = restClient
+        .post()
+        .uri(uri)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(detectionParameter)
+        .exchange(((clientRequest, clientResponse) -> {
+              ObjectMapper mapper = new ObjectMapper();
+              mapper.setSerializationInclusion(Include.ALWAYS);
+              mapper.registerModule(new JavaTimeModule());
+              mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+              if (clientResponse.getStatusCode().is2xxSuccessful()) {
+                DetectionDeBiasResult result = mapper.readValue(clientResponse.getBody(), DetectionDeBiasResult.class);
+                return new ResponseEntity<>(result, clientResponse.getStatusCode());
+              } else {
+                String errorResponse = new String(clientResponse.getBody().readAllBytes(), StandardCharsets.UTF_8);
+                if (!errorResponse.isBlank()) {
+                  ErrorDeBiasResult result = mapper.readValue(errorResponse, ErrorDeBiasResult.class);
+                  if (result.getDetailList() !=null) {
+                    throw new DeBiasBadRequestException(clientResponse.getStatusCode() + " "
+                        + result.getDetailList().getFirst().getType() + " "
+                        + result.getDetailList().getFirst().getMsg());
+                  } else {
+                    throw new DeBiasBadRequestException(errorResponse);
+                  }
+                } else if (clientResponse.getStatusCode().is5xxServerError()) {
+                  throw new DeBiasInternalServerException(clientResponse.getStatusCode().value()+" "+clientResponse.getStatusText());
+                } else if (clientResponse.getStatusCode().is4xxClientError()) {
+                  throw new DeBiasBadRequestException(clientResponse.getStatusCode().value()+" "+clientResponse.getStatusText());
+                }
+              }
+              return new ResponseEntity<>(clientResponse.getStatusCode());
+            })
+        );
+
+    if (response.getStatusCode().is2xxSuccessful()) {
+      LOGGER.info("Detection processed successfully!");
+      return response.getBody();
+    } else {
+      LOGGER.warn("Failed to process request. Response code: {}", response.getStatusCode().value());
+      return response.getBody();
+    }
```

Comment:
The return statement is the same and can be extracted from the if and else bodies and put at the very end.